src/util/util.cs

```diff
@@ -226,37 +226,52 @@ public static ProcessResult ExecuteProcess(string name, IEnumerable<string> comm
             }
 
             // Finally, start the process.
-            process.Start();
-
-            if (capture)
-            {
-                // Set up async reads for stderr and stdout
-                process.BeginErrorReadLine();
-                process.BeginOutputReadLine();
-            }
-            else
+            try
             {
-                string stdout = process.StandardOutput.ReadToEnd();
-                string stderr = process.StandardError.ReadToEnd();
+                process.Start();
 
-                if (!string.IsNullOrEmpty(stdout))
+                if (capture)
                 {
-                    Console.Write(stdout);
+                    // Set up async reads for stderr and stdout
+                    process.BeginErrorReadLine();
+                    process.BeginOutputReadLine();
                 }
-                if (!string.IsNullOrEmpty(stderr) && (stdout != stderr))
+                else
                 {
-                    Console.Write(stderr);
+                    string stdout = process.StandardOutput.ReadToEnd();
+                    string stderr = process.StandardError.ReadToEnd();
+
+                    if (!string.IsNullOrEmpty(stdout))
+                    {
+                        Console.Write(stdout);
+                    }
+                    if (!string.IsNullOrEmpty(stderr) && (stdout != stderr))
+                    {
+                        Console.Write(stderr);
+                    }
                 }
-            }
 
-            process.WaitForExit();
+                process.WaitForExit();
 
-            return new ProcessResult()
+                return new ProcessResult()
+                {
+                    ExitCode = process.ExitCode,
+                    StdOut = capture ? _outputDataStringBuilder.ToString() : string.Empty,
+                    StdErr = capture ? _errorDataStringBuilder.ToString() : string.Empty
+                };
+            }
+            catch (System.Exception e)
```

Comment:
I think it would be better to do this more locally around `process.Start()` above so that we don't end up printing confusing errors if any of the other function calls above throw an exception.